include/sway/config.h

```diff
@@ -110,12 +110,19 @@ struct bar_config {
 	 * Always visible in dock mode. Visible only when modifier key is held in hide mode.
 	 * Never visible in invisible mode.
 	 */
-	char *mode;
+	char *display_mode;
 	/**
 	 * One of "show" or "hide".
 	 *
 	 * In "show" mode, it will always be shown on top of the active workspace.
 	 */
+	char *mode;
+	/**
+	 * Current keybinding mode
+	 *
+	 * Names should probably be reworked at some point, but I'm keeping it for now
+	 * Don't think this does anything at the moment
```

Comment:
It looked like binding modes are passed directly to the bar, so this never gets changed. If instead we wanted to pass entire bar configs to the bar, we would need to handle that bar's binding modes on the sway side, so this would be necessary. In theory this would have the advantage of keeping sway up to date about the bar's state without requiring swaybar to send its state over ipc.